Improve sidecar documentation

povilasv opened the pull request.

Bplotka commented on docs/components/sidecar.md:
````diff
@@ -25,36 +25,81 @@ $ thanos query \
 ```
+
+
+## S3 configuration
````
This is perfect, just I would put it somewhere else and add short GCS configuration explanation (can help with that).

Basically sidecar is not the only guy that require this. Store, compactor, ruler and bucket comands needs this as well

povilasv replied:
:+1: how does `docs/storage-configuration.md` sounds?

Bplotka replied:
Fun fact, there is pending PR for page that can have all of this. Let me merge this one first maybe: https://github.com/improbable-eng/thanos/pull/328
ok?

Bplotka replied:
can we rebase now and use `docs/storage.md`?

povilasv replied:
:+1:

Bplotka commented on docs/storage.md:
```diff
@@ -20,4 +20,32 @@ NOTE: Currently Thanos requires strong consistency (write-read) for object store
+
+## GCP Configuration 
+
+In configure Google Cloud Storage bucket as an object store you need to set this mandatory flag:
```
or maybe you want to fill it based on https://github.com/golang/oauth2/blob/master/google/default.go#L43

povilasv replied:
Will do

povilasv commented:
@Bplotka thanks :)

deejay1 commented on docs/components/sidecar.md:
```diff
@@ -25,36 +25,58 @@ $ thanos query \
+
+## Reloader Configuration
+
+Thanos can watch changes in prometheus configuration and refresh Prometheus configuration if `--web.enable-lifecycle` enabled.
```
maybe we should keep the letter case and use `Prometheus` in both instances here?

deejay1 commented on docs/storage.md:
```diff
@@ -20,4 +20,46 @@ NOTE: Currently Thanos requires strong consistency (write-read) for object store
+
+## S3 configuration
+
+Thanos uses minio client to upload prometheus data into s3.
```
as above

deejay1 commented on docs/storage.md:
```diff
@@ -20,4 +20,46 @@ NOTE: Currently Thanos requires strong consistency (write-read) for object store
+
+and set `S3_SECRET_KEY` environment variable with AWS secret key.
+
+Instead of using flags you can pass all the configuration via environment flags:
```
`environment variables`?

deejay1 commented on docs/storage.md:
```diff
@@ -20,4 +20,46 @@ NOTE: Currently Thanos requires strong consistency (write-read) for object store
+
+## GCP Configuration 
+
+TO configure Google Cloud Storage bucket as an object store you need to set `--gcs.bucket` with GCS bucket name and configure Google Application credentials.
```
`To`

povilasv commented:
@deejay1  thanks for reviewing, I fixed those issues. 
@Bplotka don't worry I've changed a couple of things after review, so probably introduced these issues later ;)

Bplotka commented on docs/storage.md:
```diff
@@ -20,4 +20,46 @@ NOTE: Currently Thanos requires strong consistency (write-read) for object store
+
+AWS region to endpoint mapping can be found in this [link](https://docs.aws.amazon.com/general/latest/gr/rande.html#s3_region)
+
+If you are getting access denied errors try running with `--s3.signature-version2` or `--s3.insecure` flags.
```
Not sure about this line, basically, you should use HTTPS all the time and use `insecure` only for debug purposes (also how using just HTTP fixes Access Denied error)?

Maybe:
```
Make sure you use a correct signature version with `--s3.signature-version2`, otherwise, you will get Access Denied error.
```

```
For debug purposes you can `--s3.insecure` to switch to plain insecure HTTP instead of HTTPS
```
